examples/transmission_with_raman_main_example.py

```diff
@@ -0,0 +1,304 @@
+        else:
+            print(f'\nPropagating in gain mode: power cannot be set manually')
+        infos = propagate2(path, req, equipment, show=len(power_range)==1)
+        if power_mode:
+            print(f'\nTransmission result for input power = {lin2db(req.power*1e3):.2f}dBm :')
+        else:
+            print(f'\nTransmission results:')
+        #info message in gain mode
+        print(destination)
+
+        #print(f'\n !!!!!!!!!!!!!!!!!     TEST POINT         !!!!!!!!!!!!!!!!!!!!!')
+        #print(f'carriers ase output of {path[1]} =\n {list(path[1].carriers("out", "nli"))}')
+        # => use "in" or "out" parameter
+        # => use "nli" or "ase" or "signal" or "total" parameter        
+        if power_mode:
+            simulation_data.append({
+                        'Pch_dBm'               : pref_ch_db + dp_db,
+                        'OSNR_ASE_0.1nm'        : round(mean(destination.osnr_ase_01nm),2),
+                        'OSNR_ASE_signal_bw'    : round(mean(destination.osnr_ase),2),
+                        'SNR_nli_signal_bw'     : round(mean(destination.osnr_nli),2),
+                        'SNR_total_signal_bw'   : round(mean(destination.snr),2)
+                                })
+        else:
+            simulation_data.append({
+                        'gain_mode'             : 'power canot be set',
+                        'OSNR_ASE_0.1nm'        : round(mean(destination.osnr_ase_01nm),2),
+                        'OSNR_ASE_signal_bw'    : round(mean(destination.osnr_ase),2),
+                        'SNR_nli_signal_bw'     : round(mean(destination.osnr_nli),2),
+                        'SNR_total_signal_bw'   : round(mean(destination.snr),2)
+                                })          
+        #info message in gain mode
+    write_csv(result_dicts, 'simulation_result.csv')
+    return path, infos
+
+
+parser = ArgumentParser()
+parser.add_argument('-e', '--equipment', type=Path,
+                    default=Path(__file__).parent / 'eqpt_with_raman_config.json')
+parser.add_argument('-sim', '--sim-params', type=Path,
+                    default=Path(__file__).parent / 'sim_params.json', help='Path to the json containing simulation parameters')
+parser.add_argument('-pl', '--plot', action='store_true')
+parser.add_argument('-v', '--verbose', action='count', default=0, help='increases verbosity for each occurence')
+parser.add_argument('-l', '--list-nodes', action='store_true', help='list all transceiver nodes')
+parser.add_argument('-po', '--power', default=0, help='channel ref power in dBm')
+parser.add_argument('-names', '--names-matching', action='store_true', help='display network names that are closed matches')
+parser.add_argument('filename', nargs='?', type=Path,
+                    default=Path(__file__).parent / 'raman_edfa_example_network.json')
+parser.add_argument('source', nargs='?', help='source node')
+parser.add_argument('destination',   nargs='?', help='destination node')
+
```

Comment:
When my other comments are implemented, this file ends up very similar to the existing `transmission_main_example.py`. My recommendation is therefore to control selection of the Raman vs. "traditional" engine just by the choice of the input JSON file ("does it contain `RamanFiber`?").

Looking at how the code looks right now, I don't see any possible regression. We don't have that concept of "Raman amplifiers" yet, so autodesign is not affected as far as I could tell.